### DISCUSSION TITLE
Failure use case with GitHub integration regarding automatic build/deployment on commit

### DISCUSSION THREAD

**Discussion by eyolik** [vercel/community#2676] [Bug Report]
### Page to Investigate

I fixed the problem at this time

### Steps to Reproduce

The following scenario leads to failing to build/deploy a project after a commit:

Context: 
- Two GitHub accounts are involved: user1 and user2. Only user1 has Vercel installed as app in its GitHub settings.
- Vercel account is linked to user1 GitHub account, they have same email address too.
- commits are made into user1 repository using user2 name and email address (this is what is messing things up)

Steps:
1. Project is installed in Vercel (user1) and deployed successfully (all the commits were made using user2 email address/name).
2. A commit is being made, using user2 name and email address.

Expected result: 
Automatic build and deployment would be made by Vercel. If user2 was able to commit and push to the repository, Vercel should not prevent the build.

Actual result: 
Vercel fails to build/deploy. An error is visible in GitHub commit history saying:

"@user2 is attempting to deploy a commit to the user1's Team Team on Vercel.
To accomplish this, @user2 needs to request access to the Team.
Afterwards, an owner of the Team is required to accept their membership request."

Why is this case happening: 
When using GitHub Desktop, the login to git account and the name/email used for the commit are separate. We may change the git account login but forget to change the name/email used for the commit, so they may be different.

**Comment by amyegan** [comment]
Hi @eyolik. This sounds like normal behavior rather than a bug. When [deploying a private repo to a team's project](https://vercel.com/docs/concepts/deployments/git#using-teams), the commit author must be a member of the Team that owns the project. If commit author information is not appropriately configured locally, the email address mismatch prevents team membership verification.

**Reply by eyolik** [reply]
Well there must be something I didn't get then because that logic doesn't make sense to me for the following reasons:
1. all the initial commits had been made by user2 and the site was built and deployed without error. The error began when user2 made an additional commit after initial deployment.
2. if user2 makes a commit, build does not start, however if user1 makes a commit right afterwards, the build starts, and will include both user1 and user2 changes. 

The simple logic saying "you are not user of team so I don't take your commit" is not at all a security enforcement, and it can lead to issues as the one I explained above with GitHub Desktop. If you don't wish to change this behavior, at least make it clear in both the documentation and the error message which is generated when the build fails. I spent quite some time to figure out what the problem was.
